packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -929,7 +922,7 @@ export const $access: AccessDecorator = (
 ) => {
   if (typeof value.value !== "string" || (value.value !== "public" && value.value !== "internal")) {
     reportDiagnostic(context.program, {
-      code: "access",
+      code: "incorrect-access",
```

Comment:
isn't it `invalid-access` in lib.ts that should make the build fail, doesn't it?